pkg/sql/conn_executor.go

```diff
@@ -3718,6 +3720,27 @@ func (ex *connExecutor) initEvalCtx(ctx context.Context, evalCtx *extendedEvalCo
 	evalCtx.copyFromExecCfg(ex.server.cfg)
 }
 
+// GetPCRReaderTimestamp if the system database is setup as PCR
+// catalog reader, then this function will return an non-zero timestamp
+// to use for all read operations.
+func (ex *connExecutor) GetPCRReaderTimestamp(ctx context.Context) hlc.Timestamp {
+	lm := ex.server.cfg.LeaseManager
+	if lm == nil {
+		return hlc.Timestamp{}
+	}
+	desc, err := lm.Acquire(ctx, ex.server.cfg.Clock.Now(), keys.SystemDatabaseID)
+	if err != nil {
+		log.Infof(ctx, "unable to lease system database to determine if PCR reader is in use: %s", err)
+		return hlc.Timestamp{}
```

Comment:
Yeah, we could probably do this check once when we setup the conn executor and then skip it in every request if we know we're not a reader since that shouldn't really change over the lifetime of the server since we setup the reader during bootstrap (though I don't recall if a pgwire conn exec could have been started before bootstrap ran?)